### ISSUE TITLE
Search suggest

### ISSUE THREAD

**Issue by tblacker7** [nhsuk/nhsuk-frontend#568]
# Bug Report

## Search Suggest box

When a user enters text in the search box, a list of suggestions comes up. Clicking any of these takes the user to the search results on https://nhs.uk - not the site the search box originates on. The results are being generated by javascript which is pulling in from a third party service - inside nhsuk.min.js towards the end of the file is this chunk of code:
```
function f(e, t) {
        var n = "".concat("https://nhs.funnelback.co.uk/s/suggest.json", "?collection=nhs-meta&partial_query=").concat(e, "&sort=0&fmt=json++&profile=&show=").concat(10),
            r = new XMLHttpRequest;
        r.open("GET", n), r.onload = function () {
            if (200 === r.status) {
                var e = JSON.parse(r.responseText).map(function (e) {
                    return e.disp
                });
                t(e)
            }
        }, r.send()
    }
```
which refers to funnelback.co.uk and appears to be a third party indexing service. There is no way of over-riding this whilst maintaining the separation of upstream nhsuk-frontend library. I would suggest this code should be specific to nhs.uk website, not to the library itself.

## What steps are required to reproduce the issue?

Deploy a site using the library. Enter a search query into the search box. Results are shown:
![Screenshot 2019-11-20 at 09 55 06](https://user-images.githubusercontent.com/44897304/69243322-bc492200-0b9a-11ea-9457-f9ff8371ee91.png)
Clicking any of these links takes user to search page on https://nhs.uk

## What was the environment where this issue occurred?

* NHS.UK frontend package version: 3.0.2
* Node version: v11.1.0
* npm version: 6.4.1

## Is there anything else you think would be useful in recreating the issue?

 -

**Comment by AdamChrimes** [comment]
Thanks for raising this @tblacker7, the team will triage this on Monday when we get together but its something we should look into.

In the meantime you could possibly disable it by removing `<div class="autocomplete-container" id="autocomplete-container"></div>` from the Header HTML and you can change the search form action to point to your own `<form class="nhsuk-header__search-form" id="search" action="https://www.nhs.uk/search/" method="get" role="search">`

**Comment by tomdoughty** [comment]
Hi @tblacker7, if you change the IDs you can configure your own autocomplete. Using JS available from the frontend library. Just hook up your own API in `source` for the lookup and specify your own HTML in `suggestion` to be returned in the dropdown.

This has bought to my attention some CSS issues where we have styled autocompletes to belong in the header but that should be easy enough to overwrite.

HTML
```
<form id="demo-autocomplete-form" action="/">
  <div id="demo-autocomplete-container">
    <div class="nhsuk-form-group">
      <label class="nhsuk-label" for="demo-autocomplete-input">
        Enter a search term
      </label>
      <input class="nhsuk-input" id="demo-autocomplete-input" name="example" type="text">
    </div>
  </div>
</form>
```

JS
```
import AutoComplete from 'nhsuk-frontend/packages/components/header/autoCompleteConfig';

/**
 * Function to build truncated result with svg for search autocomplete
 * @param {string} result String containing individual result from autocomplete source function
 * @returns {string} String of HTML containing passed result
*/
const suggestion = (result) => `
  <svg class="nhsuk-icon nhsuk-icon__search" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" aria-hidden="true"><path d="M19.71 18.29l-4.11-4.1a7 7 0 1 0-1.41 1.41l4.1 4.11a1 1 0 0 0 1.42 0 1 1 0 0 0 0-1.42zM5 10a5 5 0 1 1 5 5 5 5 0 0 1-5-5z"></path></svg>
  ${result}
`;

/**
 * Function to populate the search autocomplete.
 * @param {string} query Query to pass to search API
 * @param {function} populateResults Callback function passed to source by autocomplete plugin
*/
const source = (query, populateResults) => {
  // Build URL for search endpoint
  fetch(`https://nhs.funnelback.co.uk/s/suggest.json?collection=nhs-meta&partial_query=${query}&sort=0&fmt=json++&profile=&show=10`)
    .then(res => res.json())
    .then(json => populateResults(json.map(({ disp }) => disp)));
};

AutoComplete({
  formId: 'demo-autocomplete-form',
  containerId: 'demo-autocomplete-container',
  inputId: 'demo-autocomplete-input',
  source,
  templates: {
    suggestion,
  }
});

```

**Comment by AdamChrimes** [comment]
Quick update on this;

We're looking at moving the search autocomplete to an internal service soon, which will remove the third party search indexing. In the near future we're going to review whether the autocomplete should be included in the frontend library or whether it should be specific to the NHS website.

**Comment by AdamChrimes** [comment]
This will be fixed in release 5.0.0 (next week)
